include/sys/arch_interface.h

```diff
@@ -675,6 +675,42 @@ extern uint64_t arch_timing_value_swap_temp;
 
 /** @} */
 
+/**
+ * @defgroup arch_cache Architecture-specific cache functions
+ * @ingroup arch-interface
+ * @{
+ */
+
+#ifdef CONFIG_CACHE_FLUSHING
+
+/**
+ *
+ * @brief Flush d-cache lines to main memory
+ *
+ * @see sys_cache_flush
+ */
+void arch_dcache_flush(void *addr, size_t size);
+
+/**
+ *
+ * @brief Invalidate d-cache lines
+ *
+ * @see sys_cache_invd
+ */
+void arch_dcache_invd(void *addr, size_t size);
+#ifndef CONFIG_CACHE_LINE_SIZE
```

Comment:
here to, add a new line on top and have the if just before the doxygen comment.

Reply:
Done. Please check if it is done correctly. Thanks